tests/test_WavelengthFitData.py

```diff
@@ -67,25 +79,38 @@ def testFromSpectrumSet(self):
         centerValues = []
         wavelengthKnots = []
         wavelengthValues = []
+        lines = []
+
         for ii, ss in enumerate(spectra):
-            ss.fiberId = self.fiberId[ii]
-            line = ReferenceLine("fake")
-            line.status = ReferenceLine.Status.FIT
-            line.wavelength = self.actualWavelength[ii]
-            line.fitPosition = self.pixels[ii]
-            ss.setReferenceLines([line])
-            knots = np.array([-1, 0, self.pixels[ii], self.length, self.length + 1], dtype=np.float32)
+            knots = np.array([-1, 0, self.fitPixelPos[ii], self.length, self.length + 1], dtype=np.float32)
             centerKnots.append(knots)
             wavelengthKnots.append(knots)
-            centerValues.append(np.array([12.34, 123.45, 234.56, 345.67, 456.78], dtype=np.float32))
-            wavelengthValues.append(np.array([self.wlMin - 1, self.wlMin, self.measuredWavelength[ii],
+            centerValues.append(self.nominalPixelPos)
+            wavelengthValues.append(np.array([self.wlMin - 1, self.wlMin, self.fitWavelength[ii],
                                               self.wlMax, self.wlMax + 1],
                                              dtype=np.float32))
+
         bbox = lsst.afw.geom.Box2I(lsst.afw.geom.Point2I(0, 0),
                                    lsst.afw.geom.Extent2I(self.length, self.length))
-        detMap = DetectorMap(bbox, self.fiberId, centerKnots, centerValues, wavelengthKnots, wavelengthValues)
+        detMap = DetectorMap(bbox, self.fiberId,
+                             centerKnots, centerValues, wavelengthKnots, wavelengthValues)
 
-        wlFitData = WavelengthFitData.fromSpectrumSet(spectra, detMap)
+        for ii, ss in enumerate(spectra):
+            ss.fiberId = self.fiberId[ii]
+            line = ReferenceLine("fake")
+            line.status = ReferenceLine.Status.FIT
+            line.wavelength = self.reflines[ii]
+            line.fitPosition = self.fitPixelPos[ii]
+            line.fitPositionErr = self.fitPixelPosErr[ii]
+            ss.setReferenceLines([line])
+
+            linedata = LineData(ss.fiberId, line.wavelength, 0, 0, 0,
+                                line.fitPosition, detMap.getWavelength(ss.fiberId, line.fitPosition),
+                                0, 0, 0, line.status)
```

Comment:
Please use actual values for the ones you've set to `0` here, and then test that they are set properly.